crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -21,6 +21,15 @@ pub enum ShaderReflectError {
     #[error(transparent)]
     Validation(#[from] naga::WithSpan<naga::valid::ValidationError>),
 }
+
+/// Describes whether or not to perform runtime checks on shaders
```

Comment:
This needs more details on the costs, risks and when you might choose each of these options.